### DISCUSSION TITLE
Any examples of buliding a macOS App Bundle?

### DISCUSSION THREAD

**Discussion by vonnieda** [slint-ui/slint#4349] [Q&A] [answered]
I'm trying to build a macOS App Bundle that I can distribute but I'm stuck somewhere between code signing and Qt bundling. Does anyone have a working example of this?

At the moment I'm trying to use cargo bundle and that gets me an .app, and I can run the contained macOS binary from the command line, but if I try to run it from Finder it just crashes immediately.

Thanks,
Jason

**Comment by tronical** [comment]
When it crashes, do you get any dialog with details about the crash?

When you run it from the command line, do you have DYLD_FRAMEWORK_PATH set by chance?

Is there a specific reason you're trying to bundle Qt? Does it work if you skip the Qt dependency?

This guy recently made a video of creating an app with Rust and Slint on macOS, which includes creating an .app bundle and launching it from Finder: https://youtu.be/7aFgeUG9TK4?si=hCy3DM4aECTHdRau

**Comment by vonnieda** [comment]
Thank you for your quick response! I think I've got this sorted. I was only using Qt as a default, now I'm trying winit-software. I had a second issue where I was trying to access a sandboxed directory. I'll report back here with results soon. Thanks!

Edit: I should note, I'm trying to prepare an app bundle for the app store. That video is helpful!

**Comment by vonnieda** [comment, marked as answer]
Alright, I think I was just confused about backend selection in general. It all builds, signs, notarizes, and runs fine.

If it helps anyone else, here's how I'm doing that:

```
#!/bin/bash

APP="../target/release/bundle/osx/Dimple.app"
BIN="$APP/Contents/MacOS/dimple_ui_slint"
ZIP="$APP/../Dimple.app.zip"
IDENT="XXXXXXXXX"
USERNAME="xxxxxxx"
PASSWORD="xxxxxxx"

cargo bundle --release
codesign --timestamp --verify -vvv --deep --options=runtime --sign $IDENT $APP
zip -r $ZIP $APP
xcrun notarytool submit --apple-id $USERNAME --team-id $IDENT --password $PASSWORD --wait $ZIP
xcrun stapler staple $APP

```
